* programs/Simulation/HDGeant/gelhad/gheishp.F [rtj]

rjones30 opened the pull request:
- make implicit integer conversions from real explicit, to avoid
   compiler warnings

rjones30 committed:
* programs/Simulation/HDGeant/gelhad/gheishp.F [rtj]
   - make implicit integer conversions from real explicit, to avoid
     compiler warnings

rjones30 committed:
* programs/Utilities/hddm/hddm-c.cpp, hddm-cpp.cpp [rtj]
   - define an enum that represents the supported attribute types in
     hddm, currently int, long, float, double, string, anyURI, Particle_t,
     and unknown (to capture all other types, eg. literal strings).
   - add optional argument to the generic getAttribute(string name)
     method so that in addition to returning a pointer to the attribute
     value (or NULL if not defined), it can also return an enum
     representing the type of the attribute (eg. "string", "float", etc.)
     This is returned iff the second argument named "atype" is passed
     as a pointer to int. Default atype is NULL, which gives the previous
     behavior where no type information is provided with the returned
     function result (pointer to void) leaving it up to the user to
     know how to typecast from void* to the appropriate c type. This
     extension is backward-compatible, but makes it possible for the
     python extension to do runtime type introspection on returned hddm
     attribute values.

gluex commented:
Build status for this pull request: FAILURE

 Build log: [/work/halld/pull_request_test/sim-recon^rtj_comp_warnings/make_rtj_comp_warnings.log](https://halldweb.jlab.org/pull_request_test/sim-recon^rtj_comp_warnings/make_rtj_comp_warnings.log)
 Build report: [/work/halld/pull_request_test/sim-recon^rtj_comp_warnings/report_rtj_comp_warnings.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^rtj_comp_warnings/report_rtj_comp_warnings.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^rtj_comp_warnings

rjones30 committed:
* programs/Simulation/HDGeant/gelhad/gheishp.F [rtj]
   - I guess gfortran 4.9.2 now no longer supports the fortran intrinsic
     int()?? So be it, henceforth it shall be called integer().

gluex commented:
Build status for this pull request: FAILURE

 Build log: [/work/halld/pull_request_test/sim-recon^rtj_comp_warnings/make_rtj_comp_warnings.log](https://halldweb.jlab.org/pull_request_test/sim-recon^rtj_comp_warnings/make_rtj_comp_warnings.log)
 Build report: [/work/halld/pull_request_test/sim-recon^rtj_comp_warnings/report_rtj_comp_warnings.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^rtj_comp_warnings/report_rtj_comp_warnings.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^rtj_comp_warnings

rjones30 committed:
* programs/Simulation/HDGeant/gelhad/gheishp.F [rtj]
   - not correct, the intrinsic for real-to-integer conversion is still
     called "int" in fortran 90. The problem was that the author is this
     function decided to use "int" as the name of one of his variables,
     thus hiding the intrinsic of the same name! I re-introduce it by
     creating a wrapper at the bottom of the file called int_intrinsic.

gluex commented:
Test status for this pull request: SUCCESS

  Summary: [/work/halld/pull_request_test/sim-recon^rtj_comp_warnings/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^rtj_comp_warnings/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/sim-recon^rtj_comp_warnings/tests/log](https://halldweb.jlab.org/pull_request_test/sim-recon^rtj_comp_warnings/tests-logs)

 Build log: [/work/halld/pull_request_test/sim-recon^rtj_comp_warnings/make_rtj_comp_warnings.log](https://halldweb.jlab.org/pull_request_test/sim-recon^rtj_comp_warnings/make_rtj_comp_warnings.log)
 Build report: [/work/halld/pull_request_test/sim-recon^rtj_comp_warnings/report_rtj_comp_warnings.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^rtj_comp_warnings/report_rtj_comp_warnings.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^rtj_comp_warnings
